### PR TITLE
BUG: pd.NA showing up as NaN in Categorical repr

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -2001,12 +2012,17 @@ def __repr__(self) -> str:
         String representation.
         """
         _maxlen = 10
+        na_repr = "NaN"
+        if is_extension_array_dtype(self.categories.dtype):
```
Comment (jbrockmendel) [inline pandas/core/arrays/categorical.py:2016]: heads up: we've moving away from is_foo_dtype, so this would be `isinstance(..., ExtensionDtype)`

Comment (lithomas1) [reply pandas/core/arrays/categorical.py]: Thanks.

**Comment by topper-123** [comment]
I have been thinking about a change to the use of the na value in `Categorical`, and just wrote up #52687 after being reminded by this PR. This would also change the repr IMO as a consequence of the change, so I  think that would make this PR unnecessary,.

**Comment by lithomas1** [comment]
Sure, happy to see what you got. 

Feel free to take my test(or any parts of this PR) if you need it.

File: pandas/tests/arrays/categorical/test_repr.py [hunk 2]
```diff
@@ -253,6 +257,19 @@ def test_categorical_repr_int_with_nan(self):
 Categories (2, int64): [1, 2]"""
         assert repr(s) == s_exp
 
+    @pytest.mark.parametrize("values_dtype", tm.ALL_INT_EA_DTYPES)
+    def test_categorical_repr_nullable_int_NA(self, values_dtype):
+        arr = array([1, 2, np.nan], dtype=values_dtype)
+        c = Categorical(arr)
+        c_exp = f"""[1, 2, {NA}]\nCategories (2, {values_dtype}): [1, 2]"""
```
Comment (jbrockmendel) [inline pandas/tests/arrays/categorical/test_repr.py:264]: so if you did `c[2]` would you get pd.NA or np.nan?  i'm wary of changing the repr without changing the item

**Comment by mroeschke** [comment]
Looks like this has gone stale. Closing to clear the queue but feel free to reopen if you want to continue

**Comment by topper-123** [comment]
FYI, there is a slightly related #53190, which isn’t getting traction though…

**Pull request closed without merging**